File: scratch/debug_json.js
const fs = require('fs');
const path = 'e:\\My Website\\Dien-Nang-Luong-Mat-Troi\\data\\db.json';
const pos = 22128;

if (fs.existsSync(path)) {
    const fd = fs.openSync(path, 'r');
    const buffer = Buffer.alloc(400);
    const start = Math.max(0, pos - 200);
    fs.readSync(fd, buffer, 0, 400, start);
    console.log(`Chunk around position ${pos} (start ${start}):`);
    console.log(buffer.toString('utf8'));
    
    for (let i = 0; i < buffer.length; i++) {
        const byte = buffer[i];
        if (byte < 32 && byte !== 9 && byte !== 10 && byte !== 13) {
            console.log(`Found control character ${byte} at absolute position ${start + i}`);
        }
    }
    fs.closeSync(fd);
} else {
    console.log("File not found");
}

File: scratch/debug_json.py
import os

file_path = r'e:\My Website\Dien-Nang-Luong-Mat-Troi\data\db.json'
pos = 22128

if os.path.exists(file_path):
    with open(file_path, 'rb') as f:
        f.seek(max(0, pos - 100))
        chunk = f.read(200)
        print(f"Chunk around position {pos}:")
        print(chunk)
        
        # Check for control characters in the chunk
        for i, byte in enumerate(chunk):
            if byte < 32 and byte not in (9, 10, 13): # Not tab, LF, CR
                print(f"Found control character {byte} at position {pos - 100 + i}")
else:
    print("File not found")
